### DISCUSSION TITLE
Ability to Apply IF-Conditional on Test Profile Data

### DISCUSSION THREAD

**Discussion by snapshotleisure** [testsigmahq/testsigma#20] [Feature Ideas]
It would be great if the IF-Conditional for the Test Case steps can be used on the Data Profiles data. Currently the IF-Conditional only apply to elements on the page. It would be great if there was a feature for "If @|total| greater than 1000", then do some action, else do some action. It would make the data-driven test cases more flexible and avoid duplication. 

However am wondering if that would make test cases too complicated at the same time, I guess it's up to the test case developer? Happy to have a discussion around this

**Comment by diveshmehta06** [comment]
Hi @sfzhao99 I am Divesh from Testsigma.
We have in our roadmap an enhancement just like this planned in Q2 (April/May).

Can we discuss a few of the use cases you have a for an enhancement like this? will help evolve the solution even better .
Please let me know

**Reply by amitsavyasachi** [reply]
Hi @sfzhao99 ,

We have this requirement covered with a couple of default addons. 

For any Numeric operation, you can use - **String Data Generators** , this uses relational operators to check if both the data are {<,>,=}.
And for String operations, you can install and use - **String Actions** , this will help you to compare two strings and also give you leverage to validate if string 1 contains string 2. String 1 and String 2 can come from any source. 

 You can view and install them here - https://testsigma.com/addons 
 
 Give me a shout if you have any questions.
